### PR TITLE
Add lines_with_custom_color_count information in metadatas

### PR DESCRIPTION
**Pull request by fchabouis**
A route is considered to have a custom color if the route color is not white or the route text color is not black.

### CODE REVIEW

**Comment by Tristramg** [comment]
This is going to be a bit tricky :grimacing: 

The GTFS specification says that if no color is provided, the color is white.

> Route color designation that matches public facing material. Defaults to white (FFFFFF) when omitted or left empty

So it was an actual bug in the GTFS library to have it as an option and it was fixed https://github.com/rust-transit/gtfs-structure/commit/2b342876bd82f04b407ad1f6d71516566e4995cb

Maybe a check would be that if all lines are white, there should be an information level saying that’s not nice (while thinking about networks with only 1 or 2 lines to avoid too much noise)

**Comment by fchabouis** [comment]
Thanks @Tristramg .
I propose to output the number of lines having custom colors in the metadata then. And by comparing to the total number of lines, users can deduce how complete the color information is.

File: src/metadatas.rs
Comment (AntoineAugusti) [inline src/metadatas.rs]: The test name is weird because you have 3 custom colors.

File: src/metadatas.rs
Comment (AntoineAugusti) [inline src/metadatas.rs]: Wondering why you expect to have the message "Failed to load data"

File: src/metadatas.rs
Comment (AntoineAugusti) [inline src/metadatas.rs:105]: lines_with_custom_color_count -> routes_with_custom_color_count?

File: src/metadatas.rs
Comment (fchabouis) [reply src/metadatas.rs]: I agree the `expect` syntax is misleading. It is the message you will get if you fail to load the GTFS data.
https://stackoverflow.com/questions/66362625/why-is-rusts-expect-called-expect

File: src/metadatas.rs
Comment (fchabouis) [reply src/metadatas.rs:105]: I hesitated, but there is already `lines_count`, so I picked this name to be coherent.

File: src/metadatas.rs
Comment (AntoineAugusti) [reply src/metadatas.rs]: Thanks!

File: src/metadatas.rs
Comment (AntoineAugusti) [reply src/metadatas.rs:105]: Ah, understood. Coherence > adhering to the GTFS spec.

File: src/metadatas.rs
Comment (fchabouis) [reply src/metadatas.rs]: Thanks, I changed. I didn't update the test name after @Tristramg feedback.

**Review by AntoineAugusti — APPROVED** [review]
LGTM but waiting for @Tristramg's approval

File: src/metadatas.rs
Comment (Tristramg) [reply src/metadatas.rs:105]: sorry for that :speak_no_evil:

**Review by Tristramg — APPROVED** [review]

**Pull request merged**
